### ISSUE TITLE
0.76 - unable to set due time in some cases

### ISSUE THREAD

**Issue by therampagerado** [wekan/wekan#1502]
## Issue
**Impacted version**: 0.76

**Server Setup Information**:

* Operating System: Ubuntu
* Deployment Method(tar/sandstorm/snap/docker/etc /http frontend (nginx, httpd, ...)): snap
* Node Version: 8.9.3
* MongoDB Version: 
* ROOT_URL environment variable (Is there a subfolder?): no

**Problem description**:
- We're located in Bulgaria where we don't use AM/PM. When I create new card the holder text in the time input field is set to hh.mm but on some of our computers it's showing hh.mm AM. They are incorrectly set up but when I fix that the holder text is the same and I'm not able to change the due time. I can only edit until 12:00 AM and when I input 15:00 PM it does not work - invalid date/time. Those users are not admins but I've tried elevating their rights with no success.

It's important for us us because sometimes we have orders that need to have specific time set.

Also - how can I set up default time for this field? I would like to be 15:00 in our case not 12:00. This will save one click per card creation.

**Comment by xet7** [comment]
@therampagerado 

There are 2 alternatives for using time without AM/PM:

a) Change language to "English (UK)"

b) Translate Wekan to Bulgarian language here, and change to use it:
https://www.transifex.com/wekan/wekan

There is no feature for setting default time for time field yet.

**Comment by therampagerado** [comment]
With my profile and my PC I see it as hh.mm and on the rest of computers with their profiles they see it as hh.mm AM/PM.

All setups are English (no changes made).

I'm thinking of making a translation first thing when we implement it in our company but first things first - we need to start so I can have time for the translation.

**Comment by therampagerado** [comment]
I've just noticed that you changed the title of my PR.

I was reporting a bug and just shared my experience. I have a problem where I can't set the proper time in the field. I can't set 15:00 PM I can only set AM hours and I need 1500?

I'll try switching between UK and US English but I think they are all set on UK.

**Comment by xet7** [comment]
@therampagerado 

This is issue, and those are usually about bugs or feature requests. I do often change issue titles to describe what new is actually asked, it's very normal thing to do.

This is not PR = pull request, you have no new code added in this. I do also often change pull request titles, and ask for changes in text [like in this pull request that has code](https://github.com/wekan/wekan/pull/1470), because usually text can be changed to better describe what feature is implemented, so it is understood better. It's very normal thing to do.

Only new thing you are asking is setting default time.

There is no bug in setting time. If you are using US English, there is no such thing like 1500, there is only 03:00 PM, it's 12 hour clock, not 24 hour.

![wekan-start-date-us](https://user-images.githubusercontent.com/15545/36568697-a39931ac-1833-11e8-9922-a1c02081cd29.png)

When English UK is selected:
![wekan-lang-for-time](https://user-images.githubusercontent.com/15545/36568722-b47b52ac-1833-11e8-9f46-51d855d96f34.png)

Then you can add start time like 15:00
![wekan-start-date](https://user-images.githubusercontent.com/15545/36568756-d322755a-1833-11e8-8748-5fca1507547f.png)

And due date like 18:00
![wekan-due-date](https://user-images.githubusercontent.com/15545/36568772-e8e7c11a-1833-11e8-9c2f-23dde381c347.png)

**Comment by saschafoerster** [comment]
I just checked, which default times are used in the different time fields of Wekan: 

- For Received, Start and End Wekan automatically uses the actual time of creation of the fields. 
- For Due Wekan uses 17:00, (at least in our installation, maybe this field is related to the configuration field NOTIFY_DUE_AT_HOUR_OF_DAY=8, or if this only changes the time of e-mails sent).

Wouldn't it make sense to unify the behavior? Like to always use the actual time of creation? Or the set a  standard time for all fields?

<img width="513" alt="Bildschirmfoto 2019-11-16 um 10 01 11" src="https://user-images.githubusercontent.com/1153741/68990854-643bb400-0858-11ea-99a9-a6590dd30c93.png">
<img width="308" alt="Bildschirmfoto 2019-11-16 um 09 45 48" src="https://user-images.githubusercontent.com/1153741/68990855-64d44a80-0858-11ea-978d-af99fffc1ada.png">

**Comment by b90g** [comment]
> @therampagerado
> 
> This is issue, and those are usually about bugs or feature requests. I do often change issue titles to describe what new is actually asked, it's very normal thing to do.
> 
> This is not PR = pull request, you have no new code added in this. I do also often change pull request titles, and ask for changes in text [like in this pull request that has code](https://github.com/wekan/wekan/pull/1470), because usually text can be changed to better describe what feature is implemented, so it is understood better. It's very normal thing to do.
> 
> Only new thing you are asking is setting default time.
> 
> There is no bug in setting time. If you are using US English, there is no such thing like 1500, there is only 03:00 PM, it's 12 hour clock, not 24 hour.
> 
> ![wekan-start-date-us](https://user-images.githubusercontent.com/15545/36568697-a39931ac-1833-11e8-9922-a1c02081cd29.png)
> 
> When English UK is selected:
> ![wekan-lang-for-time](https://user-images.githubusercontent.com/15545/36568722-b47b52ac-1833-11e8-9f46-51d855d96f34.png)
> 
> Then you can add start time like 15:00
> ![wekan-start-date](https://user-images.githubusercontent.com/15545/36568756-d322755a-1833-11e8-8748-5fca1507547f.png)
> 
> And due date like 18:00
> ![wekan-due-date](https://user-images.githubusercontent.com/15545/36568772-e8e7c11a-1833-11e8-9c2f-23dde381c347.png)

the demonstration of changing the language & therefor the time format doesnt work in my snap craft install. although the language  in the interface is changing (from german to english UK ). am i  missing something? its behind an apache reverse proxy.

**Comment by suncobran** [comment]
Hi!
Is there a way to manually set the default card due time?

**Comment by xet7** [comment]
@suncobran 

Yes, I just tried. After selecting date, I clicked time field, and wrote different time.
